lib/internal/util.js

```diff
@@ -36,6 +37,35 @@ function objectToString(o) {
 // each one once.
 const codesWarned = {};
 
+// Returns a function that emits a deprecation on first run
+// For internal use only. `code` is a required argument.
+// If --no-deprecation is set, then it is a no-op.
+// { pending: true } are visible only when pendingDeprecation is on
+function deprecator(msg, code, options = {}) {
+  if (process.noDeprecation === true) {
+    return () => {};
+  }
+  if (options.pending && !pendingDeprecation) {
```

Comment:
Nit: as this if statement does the same as the one above I would just combine these two.